FINRA (Independent Publisher)

krautrocker opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

krautrocker committed:
- Created readme.md file for a proposed connector, FINRA

vmanoharas reviewed (commented):
Hello @krautrocker, 

Thank you very much for your proposal, kindly check this document and prepare your connector. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission-ip

krautrocker committed:
- Added file apiDefinition.swagger.json

- Added file apiProperties.json

- Added detail on the free endpoints available to the public, Equity and Fixed Income.

vmanoharas commented on independent-publisher-connectors/Proposal-Financial Industry Regulatory Authority FINRA (Independent Publisher)/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,176 @@
+  "swagger": "2.0",
+  "info": {
+    "title": "FINRA",
+    "description": "...",
```
Kindly add the description.

krautrocker replied:
This was text from another connector, some placeholder content that's since been removed.

vmanoharas commented on independent-publisher-connectors/Proposal-Financial Industry Regulatory Authority FINRA (Independent Publisher)/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,176 @@
+      "post": {
+        "responses": {
+          "200": {
+            "description": "default",
```
kindly update description, this should not be default

krautrocker replied:
This was text from another connector, placeholder content that's been removed.

vmanoharas commented on independent-publisher-connectors/Proposal-Financial Industry Regulatory Authority FINRA (Independent Publisher)/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,176 @@
+                  "items": {
+                    "type": "string"
+                  },
+                  "description": "A list of strings upon which to stop generating. You probably want [\"\n\nHuman:\"], as that's the cue for the next turn in the dialog agent.",
```
Kindly update this "[\"\n\nHuman:\"]"

krautrocker replied:
This was text from another connector, placeholder content that's been removed.

vmanoharas reviewed (commented):
Hello @krautrocker, 

Kindly fix the review comments and remove Proposal from the title. 

Additionally, please correct the folder hierarchy (.gitignore is not needed) 

![image](https://github.com/microsoft/PowerPlatformConnectors/assets/137378994/c6e10d7f-3d82-4906-9438-edd22e90ce5b)

vmanoharas commented:
Hello @krautrocker, 

I am following up on my review comments, kindly address the issues.

krautrocker commented:
@microsoft-github-policy-service agree [company="{Dan Romano aka IDR, Inc. dba SWOLCAT}"]

krautrocker committed:
#3556 - Added description to apiDefinition.

krautrocker commented:
@microsoft-github-policy-service agree

krautrocker commented:
@microsoft-github-policy-service agree company="Dan Romano"

krautrocker commented:
@microsoft-github-policy-service agree company="Dan Romano aka IDR, Inc. dba SWOLCAT"

krautrocker committed:
#3556 Retooled template apiDefinition and apiProperties json files to work for FINRA

krautrocker committed:
#3556 Updated connector description.

krautrocker committed:
Remove .gitignore file

vmanoharas commented on independent-publisher-connectors/Financial Industry Regulatory Authority FINRA (Independent Publisher)/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,329 @@
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "FINRA Custom Connector for Power Platform",
```
Please update the title according to the link below. Please remove "Power Platform" and ensure that the title should be within 30 characters. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#give-your-connector-a-title

vmanoharas commented on independent-publisher-connectors/Financial Industry Regulatory Authority FINRA (Independent Publisher)/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,329 @@
+  "info": {
+    "version": "1.0.0",
+    "title": "FINRA Custom Connector for Power Platform",
+    "description": "The FINRA Custom Connector is designed to integrate with the Financial Industry Regulatory Authority (FINRA) API, allowing seamless access to various market data endpoints. This connector supports OAuth2 authentication to ensure secure access to the FINRA data services. Users can retrieve detailed information on equity and fixed income markets, including weekly and monthly sum ...
```
The description should not cross more than 500 characters, please refer the below document. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector

vmanoharas commented on independent-publisher-connectors/Financial Industry Regulatory Authority FINRA (Independent Publisher)/readme.md:
```diff
@@ -0,0 +1,192 @@
+				- Key highlights:
+					- Weekly Treasury Aggregates - TRACE reported trading volume in US Treasury Securities aggregated weekly by security subtype.
+					- Market Breadth & Sentiment - TRACE fixed income market breadth and sentiment data.
+					- Capped Volume - TRACE structured product and coporate/agency capped volume data.
```
please correct this spelling "coporate"

vmanoharas commented on independent-publisher-connectors/Financial Industry Regulatory Authority FINRA (Independent Publisher)/readme.md:
```diff
@@ -0,0 +1,192 @@
+## Financial Regulatory Authority (FINRA) Connector
```
Please follow the indentation as per the below link. 

https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/Independent%20Publisher/readme.md

vmanoharas reviewed (commented):
Please resolve all the review comments to proceed further.

krautrocker committed:
Update .gitignore and remove unnecessary files

krautrocker committed:
Update .gitignore to exclude Visual Studio and other unnecessary files

krautrocker committed:
#3556 Removed .gitignore, updated title

krautrocker committed:
#3556 Fixed 'coporate' typo

krautrocker committed:
#3556 Re-did readme file, fixed indentation to match example from link provided.

krautrocker commented:
@vmanoharas Thanks for the help thus far. I am currently validating the connector using paconn. I haven't checked that box yet above, currently working on this. May be a day or two.

krautrocker committed:
- Modified swagger to align with paconn validate, no errors or warnings

krautrocker commented:
Screenshots from the Test operations section within the Custom Connector UI for each action:

![Custom Connector UI - Test - Response - Equity Monthly Summary](https://github.com/user-attachments/assets/bd67df7a-1b8b-4658-b883-711025598213)
![Custom Connector UI - Test - Request - Equity Monthly Summary](https://github.com/user-attachments/assets/b8d17d90-ee0a-4330-86c3-52a39a83508a)
![Custom Connector UI - Test - Request - Equity Weekly Summary](https://github.com/user-attachments/assets/f3986a1a-050e-41cd-9cc5-d79367a7a906)
![Custom Connector UI - Test - Response - Equity Weekly Summary](https://github.com/user-attachments/assets/55bf2d1d-ac0b-469f-930c-a7b4c6ff19ce)
![Custom Connector UI - Test - Request - OTC Block Summary](https://github.com/user-attachments/assets/c25e170d-e4be-47d9-8767-db37997d1e51)
![Custom Connector UI - Test - Response - OTC Block Summary](https://github.com/user-attachments/assets/2b06a0db-9e5f-4b83-b40f-8ac31ab6fe00)

krautrocker commented:
Screenshots of 3 unique operations (actions/triggers) working within a Flow:

![Custom Connector Actions - Success with Downloaded Output](https://github.com/user-attachments/assets/66153141-f94b-4e7c-a4ee-a015b8440450)
![Custom Connector Actions - Success with Detail](https://github.com/user-attachments/assets/41131bde-cbb5-4fa0-bf81-183dc59b4068)
![Custom Connector Actions - Success](https://github.com/user-attachments/assets/7c52fbf1-1e68-4f1f-bbd0-df2a355361a9)

krautrocker commented:
@vmanoharas Please let me know if anything else is required or a tweak needs to be made. Thanks.

vmanoharas commented:
Hello @krautrocker, 

Thank you very much for working with us!

Could you please remove gitignore file and checkin directly to independent-publisher-connectors folder?

Additionally, are you submitting two connectors? if so, kindly submit two different PRs, please refer below screenshot. 

![image](https://github.com/user-attachments/assets/51ebba62-750f-4b42-a8db-449818cc68f8)

krautrocker committed:
- Removed .gitignore

krautrocker commented:
@vmanoharas Of course. I removed the .gitignore and the only PR that I should have open is for FINRA (Independent Publisher). Probably happened when I changed the name.

vmanoharas commented:
Hello @krautrocker, 

It is not removed the .gitignore folder and still there are two connectors, please refer the below screen shot. 

![image](https://github.com/user-attachments/assets/04d4c13f-4230-4a17-a4ba-e3aad50fb09c)

krautrocker committed:
#3556 Removed Financial

krautrocker committed:
#3556 Removed dupe connector and .gitignore, adjusted readme.md to include further instruction

krautrocker commented:
@vmanoharas Ahhh, I see now. I removed the old connector with the longer name. That also removed the .gitignore file. Let me know if that doesn't do the trick.

krautrocker commented:
@vmanoharas Did this take care of it?

vmanoharas commented:
Hello @krautrocker, 

Unfortunately, gitignore folder is still present, kindly check the below image. 

![image](https://github.com/user-attachments/assets/c4a6fae7-d240-462a-996f-483db6bf46c2)

krautrocker committed:
#3556 Deleted .vs folder

krautrocker committed:
Delete .gitignore

krautrocker committed:
#3556 Removed objects

krautrocker committed:
Delete independent-publisher-connectors/.vs directory

krautrocker commented:
@vmanoharas Sorry about this, could you check again please? Just tried something. Thanks a lot.

vmanoharas reviewed (dismissed):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

vmanoharas commented:
[[certify-connector]]

krautrocker commented:
Hi @vmanoharas May I ask the status of deployment? Not a problem if it's taking longer than usual, just following up to see if there is anything else that I need to do.

vmanoharas commented:
Hello @krautrocker, Apologies for the delayed reply, we have an issue internally, team is actively working on this fix.  Please allow us a weeks' time.

krautrocker commented:
> Hello @krautrocker, Apologies for the delayed reply, we have an issue internally, team is actively working on this fix.  Please allow us a weeks' time. 

Hi @vmanoharas, following up but no pressure. Please let me know if there's anything I can do to help.

vmanoharas commented:
> > Hello @krautrocker, Apologies for the delayed reply, we have an issue internally, team is actively working on this fix.  Please allow us a weeks' time.
> 
> Hi @vmanoharas, following up but no pressure. Please let me know if there's anything I can do to help.

Hello @krautrocker, 

Apologies for the delay, the issue is still not fixed.  Team is working to fix the issue.  It would take another week to fix the issue.

krautrocker commented:
@vmanoharas Hey, any update here? Do I need to review my API at all? Is that causing the issue?

vmanoharas commented:
Hello @krautrocker, 

Due to the upcoming holiday season, effective from Nov 20th until January 6, 2025, we will be observing a deployment freeze to deploy the connectors. This is to ensure stability and reliability for the connector lifecycle is maintained during this period.

krautrocker commented:
@vmanoharas That's understandable and expected. I'll reach out again after the 6th.

vmanoharas commented:
[like]  Manohara S (SONATA SOFTWARE NO... reacted to your message:
________________________________
From: Dan Romano ***@***.***>
Sent: Tuesday, December 24, 2024 2:55:17 PM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Manohara S (SONATA SOFTWARE NORTH AMERICA) ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] FINRA (Independent Publisher) (PR #3556)


@vmanoharas<https://github.com/vmanoharas> That's understandable and expected. I'll reach out again after the 6th.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/PowerPlatformConnectors/pull/3556#issuecomment-2561206360>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BAYDZMXCNC6YT4CUQ4FPYBD2HFYVLAVCNFSM6AAAAABKAK5GFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRRGIYDMMZWGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

vmanoharas commented on independent-publisher-connectors/FINRA (Independent Publisher)/apiProperties.json:
```diff
@@ -0,0 +1,65 @@
+      }
+    },
+    "iconBrandColor": "#da3b01",
+    "scriptOperations": [
```
Can you please add the script file as per this property? if it is added by mistake can you please remove this section?

vmanoharas reviewed (commented):
Hello @krautrocker, 

While processing for deployment your connector, our team found one comment.  Could you please address this?

krautrocker committed:
- Removed script operations snippet, not meant to be there

- Changed stackOwner

vmanoharas reviewed (approved):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

krautrocker commented:
Thank you!

vmanoharas commented:
Hello Partner,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short survey - https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.
